zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/AuthorizationCheckBehaviorTest.java

```diff
@@ -158,6 +158,68 @@ public void shouldGetResourceIdentifiersWhenRoleHasPermissions() {
     assertThat(resourceIdentifiers).containsExactlyInAnyOrder(resourceId1, resourceId2);
   }
 
+  @Test
+  public void shouldBeAuthorizedForTenant() {
+    // given
+    final var userKey = createUser();
+    final var resourceType = AuthorizationResourceType.DEPLOYMENT;
+    final var permissionType = PermissionType.DELETE;
+    final var resourceId = UUID.randomUUID().toString();
+    addPermission(userKey, resourceType, permissionType, resourceId);
+    final var tenantId = createAndAssignTenant(userKey);
+    final var command = mockCommand(userKey);
+
+    // when
+    final var request =
+        new AuthorizationRequest(command, resourceType, permissionType, tenantId)
+            .addResourceId(resourceId);
+    final var authorized = authorizationCheckBehavior.isAuthorized(request);
+
+    // then
+    assertThat(authorized.isRight()).isTrue();
+  }
+
+  @Test
+  public void shouldBeUnauthorizedForTenant() {
+    // given
+    final var userKey = createUser();
+    final var resourceType = AuthorizationResourceType.DEPLOYMENT;
+    final var permissionType = PermissionType.DELETE;
+    final var resourceId = UUID.randomUUID().toString();
+    addPermission(userKey, resourceType, permissionType, resourceId);
+    final var anotherTenantId = "authorizedForAnotherTenant";
+    createAndAssignTenant(userKey);
+    final var command = mockCommand(userKey);
+
+    // when
+    final var request =
+        new AuthorizationRequest(command, resourceType, permissionType, anotherTenantId)
+            .addResourceId(resourceId);
+    final var authorized = authorizationCheckBehavior.isAuthorized(request);
+
+    // then
+    assertThat(authorized.isRight()).isFalse();
+  }
+
+  @Test
+  public void shouldGetAuthorizedTenantIds() {
```

Comment:
❌ Some scenarios aren't covered:

1. We expect an empty list if we can't find the user key on the command
2. We expect an empty list if we can't find the user in the state